web/src/components/tasks/business-formation/NexusFormation.test.tsx

```diff
@@ -310,6 +311,106 @@ describe("<NexusFormationFlow />", () => {
     let page: FormationPageHelpers;
 
     describe("when feature flag is set", () => {
+      describe("remembers formation step", () => {
+        const getNexusBusiness = ({
+          lastVisitedPageIndex,
+          businessName,
+        }: {
+          lastVisitedPageIndex: number;
+          businessName?: string;
+        }): Business => {
+          const legalStructureId = "limited-liability-company";
+          const profileData = generateFormationProfileData({
+            legalStructureId,
+            businessPersona: "FOREIGN",
+            businessName: businessName ?? "",
+            nexusDbaName: "",
+            needsNexusDbaName: false,
+          });
+          const formationData = generateFormationData({ lastVisitedPageIndex });
+          displayContent = {
+            formationDbaContent: generateFormationDbaContent({}),
+          };
+          return (initialBusiness = generateBusiness({ profileData, formationData }));
+        };
+
+        beforeEach(() => {
+          jest.resetAllMocks();
+          useSetupInitialMocks();
+        });
+
+        it("shows the nexus name step with an initial user when lastVisitedPage has index 0 which is by default", async () => {
+          preparePage({ business: getNexusBusiness({ lastVisitedPageIndex: 0 }), displayContent });
+          expect(screen.getByTestId("nexus-name-step")).toBeInTheDocument();
+        });
+
+        it("shows the last formation step when lastVisitedPage has step index 4", () => {
+          preparePage({
+            business: getNexusBusiness({ lastVisitedPageIndex: 4, businessName: "fake-business-name" }),
```

Comment:
I don't necessarily think the test needs to explicitly state it needs a businessName because it's part of the setup in order to get to the page? It's a precursor to what we are testing which is that when the userData has a 4, the rendered page is an index of 4. The user can't have a 4 in the userData without having entered a businessName already.